File: app/src/main/java/ru/p3tr0vich/fuel/ImplementException.java
package ru.p3tr0vich.fuel;

import android.content.Context;
import android.support.annotation.NonNull;

import java.util.Arrays;

class ImplementException  extends ClassCastException {

    ImplementException(@NonNull Context context, @NonNull Class[] ints) {
        super(context.getClass().getSimpleName() + " must implement " + Arrays.toString(ints));
    }

    ImplementException(@NonNull Context context, @NonNull Class cls) {
        this(context, new Class[]{cls});
    }
}

File: app/src/main/kotlin/ru/p3tr0vich/fuel/ImplementException.kt
package ru.p3tr0vich.fuel

import android.content.Context

internal class ImplementException(context: Context, ints: Array<Class<*>>) :
        ClassCastException(context.javaClass.simpleName + " must implement " + ints.contentToString()) {
    constructor(context: Context, cls: Class<*>) : this(context, arrayOf<Class<*>>(cls))
}